foxglove_bridge_base/src/test/test_client.cpp

```diff
@@ -132,7 +132,35 @@ std::future<Channel> waitForChannel(std::shared_ptr<ClientInterface> client,
         }
       }
     });
+  return future;
+}
+
+std::future<FetchAssetResponse> waitForFetchAssetResponse(std::shared_ptr<ClientInterface> client) {
+  auto promise = std::make_shared<std::promise<FetchAssetResponse>>();
```

Comment:
nit: is make_shared required?

Reply:
You left a similar comment here: https://github.com/foxglove/ros-foxglove-bridge/pull/239#discussion_r1253489103

Will create an issue to keep track of this